engine/include/Engine/Core/Rendering/Renderer/IRenderer.hpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Core/Debug/Assert.hpp"
+#include "Engine/Core/Rendering/Window/IWindow.hpp"
+#include <memory> //std::unique_ptr
+
+namespace Engine::Core::Renderering
+{
+class IRenderer
+{
+  protected:
+    IWindow* m_pWindow = nullptr;
+
+  public:
+
+    IRenderer(IWindow * pWin) noexcept 
+        : m_pWindow {pWin}
+    {
+    //    GPE_ASSERT(m_pWindow != nullptr);
+    }
```

Comment:
I think I accidentally removed a comment on these lines during my previous review.

What would you think of using a reference instead of a pointer on `IWindow`?
If could prevent a `nullptr` from being passed, and would require no safety checks, while keeping polymorphism.